src/coreclr/src/pal/src/thread/thread.cpp

```diff
@@ -2808,7 +2808,7 @@ PAL_InjectActivation(
         palError = InjectActivationInternal(pTargetThread);
     }
 
-    if (palError == NO_ERROR)
+    if (palError != NO_ERROR)
```

Comment:
Also found this in release/3.1. Probably it's not so important to backport? @janvorli

Reply:
It doesn't have any effect on the runtime and we backport only issues that are causing crashes that cannot be worked around or security fixes.